homeassistant/components/monzo/sensor.py

```diff
@@ -0,0 +1,129 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, cast
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import ACCOUNTS, CONF_COORDINATOR, DOMAIN, MODEL_POT, POTS
+from .entity import MonzoBaseEntity
+
+
+@dataclass
+class MonzoSensorEntityDescriptionMixin:
+    """Adds fields for Monzo sensors."""
+
+    value: Callable[[dict[str, Any]], Any]
+
+
+@dataclass
```

Comment:
Data classes should now be frozen. You can remove the Mixin by using `kw_only=True` checkout Withings on how that works